### PR TITLE
GUM-324: Return real asset ID for duplicate uploads

### PR DESCRIPTION
**Pull request by ternarybits**
## Summary
- Fix `POST /assets` to return the real asset ID (not a placeholder UUID) when a duplicate is detected
- Buffered path: use `with_raw_response.create()` to detect HTTP 200 (duplicate) vs 201 (new) from photos-api
- Streaming path: propagate HTTP status code from the pipeline so the adapter can detect duplicates
- Remove dead "duplicate"/"already exists" string-matching branch from `_handle_upload_error` — photos-api returns HTTP 200 for duplicates (not errors), so this code path was unreachable

## Linear
https://linear.app/gumnut-ai/issue/GUM-324/post-assets-duplicate-response-returns-placeholder-uuid-instead-of

## Test plan
- [x] Buffered path: new asset returns 201 + "created" status
- [x] Buffered path: duplicate returns 200 + "duplicate" status with real asset ID
- [x] Streaming path: duplicate returns 200 + "duplicate" status with real asset ID
- [x] All 647 tests pass
- [x] Pyright type check clean, ruff lint/format clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by ternarybits** [comment]
## Feedback addressed

### Auto-fixed (two-way door)
- **Pipeline instance attribute** (@charliecreates): Replaced `_http_status` dict injection with `pipeline.last_status_code` instance attribute, mirroring existing `refreshed_token` pattern. Eliminates dict mutation and key collision concerns.
- **Fail-fast on non-dict** (@charliecreates): The non-dict wrapping is no longer needed since we don't inject into the dict at all. The pipeline now returns `response.json()` unmodified.
- **Test robustness** (@charliecreates): Replaced raw byte assertions with `json.loads(result.body)` dict comparisons.

### Skipped
- "Buffered path should gate on 201" — SDK only returns for 2xx; photos-api only uses 200/201 for this endpoint. Extra gating adds complexity for an impossible scenario.
- "Streaming `UUID(int=0)` fallback" — defensive fallback for broken upstream. If photos-api returns 200 without `id`, something is deeply wrong upstream. The fallback is harmless.

File: services/streaming_upload.py [hunk 1]
```diff
@@ -288,7 +288,12 @@ def _sync_upload(
         if new_token:
             self.refreshed_token = new_token
 
-        return response.json()
+        result = response.json()
+        if isinstance(result, dict):
+            result["_http_status"] = response.status_code
+        else:
+            result = {"_http_status": response.status_code, "data": result}
+        return result
```
[the remark this answers is not in the document]
Comment (ternarybits) [reply services/streaming_upload.py]: Fixed — the non-dict wrapping is no longer needed. We moved the status code to `pipeline.last_status_code` instead of injecting into the response dict, so `response.json()` is returned unmodified.

[the remark this answers is not in the document]
Comment (ternarybits) [reply services/streaming_upload.py]: Fixed — adopted this suggestion. Status code is now stored on `pipeline.last_status_code`, mirroring the existing `refreshed_token` pattern. No dict mutation or pop needed.

File: tests/unit/api/test_assets.py [hunk 2]
```diff
@@ -448,16 +463,13 @@ async def test_upload_asset_duplicate(self, sample_uuid, mock_current_user):
 
         assert isinstance(result, JSONResponse)
         assert result.status_code == 200
-        assert (
-            result.body
-            == b'{"id":"00000000-0000-0000-0000-000000000000","status":"duplicate"}'
-        )
+        assert result.body == f'{{"id":"{sample_uuid}","status":"duplicate"}}'.encode()
```
[the remark this answers is not in the document]
Comment (ternarybits) [reply tests/unit/api/test_assets.py]: Fixed — using `json.loads(result.body)` and comparing dicts now.

**Comment by ternarybits** [comment]
## Additional fix: fail fast on streaming contract violations

Addressing charliecreates feedback that was previously skipped:

- **Missing asset ID on duplicate (200)**: Now raises `502 Bad Gateway` instead of silently returning placeholder `UUID(int=0)`. If photos-api returns 200 without an `id`, that's an upstream contract violation.
- **Missing `last_status_code`**: Now raises `502 Bad Gateway` instead of silently defaulting to 201. If the pipeline doesn't capture the status code, that's a regression.

Both have dedicated tests.

**Review by taggartgorman — APPROVED** [review]
## Pre-approve checklist
- [x] **Findings**: 0 Must Fix / 0 Should Fix (2 Consider items below)
- [x] **Security/auth**: PR does not touch authentication, authorization, or permission logic
      Evidence: Diff limited to assets.py upload paths, streaming_upload.py, and tests. No changes to auth_middleware, jwt handling, or permission checks.
- [x] **Payments/billing**: PR does not touch payments, billing, or subscription logic
      Evidence: No payment/billing code in this repo.
- [x] **Data deletion**: PR does not touch destructive operations or data removal
      Evidence: Changes are in upload (POST) path only. No DELETE endpoints modified.
- [x] **Confidence**: Correctness is well understood
      Rationale: HTTP status code detection is more reliable than string matching. Streaming path has explicit fail-fast guards (502 for missing status code or missing asset ID). Both paths covered by tests. `map_gumnut_error` correctly forwards SDK exceptions with `status_code` attributes, so 413/415 handling is preserved.
- [x] **Purpose clarity**: PR purpose and approach are clear from description and code

## Consider (non-blocking)

1. **`raw_response.parse()` outside Sentry span** — The `.parse()` call happens after the `start_span` block closes, so parsing time isn't measured. Negligible for a small JSON receipt, but could be moved inside the span for completeness.

2. **Streaming path implicit fallthrough for non-200** — Only 200 and 201 reach the routing logic (enforced by `_sync_upload`), but an explicit `elif http_status == 201` with a warning in the `else` branch would make the contract self-documenting against future upstream changes.

Clean implementation. Status-code detection is a solid upgrade over string matching. Good test coverage including edge cases (missing status code, missing asset ID on duplicate). LGTM.
